scripts/build.py

```diff
@@ -81,30 +99,101 @@ def process_third_party_libs():
 def _build_files():
     ensure_directory_exists(OUT_DIR)
     shutil.rmtree(OUT_DIR)
+    ensure_directory_exists(ASSETS_OUT_DIR)
+    shutil.rmtree(ASSETS_OUT_DIR)
+    ensure_directory_exists(THIRD_PARTY_GENERATED_OUT_DIR)
+    shutil.rmtree(THIRD_PARTY_GENERATED_OUT_DIR)
+    ensure_directory_exists(CSS_OUT_DIR)
+    shutil.rmtree(CSS_OUT_DIR)
     process_third_party_libs()
 
     for root in os.listdir(os.path.join(os.getcwd())):
-        if any([s in root for s in ['.git', 'third_party', 'extensions']]):
+        if any([s in root for s in ['.git', 'extensions', 'build']]):
             continue
 
         print 'Processing %s' % os.path.join(os.getcwd(), root)
-        for root, dirs, files in os.walk(os.path.join(os.getcwd(), root)):
-            for directory in dirs:
-                print 'Processing %s' % os.path.join(root, directory)
-            for filename in files:
-                source_path = os.path.join(root, filename)
-                if source_path.find(OUT_DIR) > 0:
+
+        # Process assets, copy it to build/[cache_slug]/assets
+        if any([s in root for s in ['assets']]):
```

Comment:
I know the old code did a bit of a walkthrough, but could we do something a bit more direct? I.e. just go to the relevant directory directly, and process all the files in it that need to be processed? Ditto below.